### PR TITLE
Use consistent pattern for namespace of extension methods

### PR DESCRIPTION
**Pull request by alanwest**
This is an alternative to #1541.

I'm beginning to lean more towards leaving things alone for the most part. That is, I think the general rule of thumb that has been followed is to place extension methods in the same namespace as the type that they are extending. This is what Microsoft has followed with many of their libraries (e.g., Microsoft.Extensions.Configuration, Microsoft.Extensions.Logging, etc.).

However, there are a few places that do not follow this rule. I believe some of them are intentional exceptions to the rule, but some of them may have been unintentional.

I this PR, I've tracked down all of the public extension methods that are _not_ contained in the same namespace as the type they are extending and changed them to be in the same namespace. There were two notable exceptions that I think are sensible exceptions to this general rule - I added a comment explaining why - please confirm the accuracy of my comments.

Questions:

* Does this seem like a good pattern to follow?
* Are there any namespace changes in this PR that anyone feels should be an exception to the pattern?

### CODE REVIEW

File: src/OpenTelemetry.Api/ActivityContextExtensions.cs [hunk 1]
```diff
@@ -16,6 +16,11 @@
 
 using System.Diagnostics;
 
+// The ActivityContext class is in the System.Diagnostics namespace.
+// These extension methods on ActivityContext are intentionally not placed in the
+// same namespace as Activity to prevent name collisions in the future.
+// The OpenTelemetry namespace is used because ActivityContext applies to all types
+// of telemetry data - i.e. both traces and metrics.
```
Comment (cijothomas) [inline src/OpenTelemetry.Api/ActivityContextExtensions.cs:23]: traces, logs and metrics

**Review by cijothomas — APPROVED** [review]
This looks good to me. We want to add this to changelog before merge.

File: src/OpenTelemetry.Exporter.Console/.publicApi/net452/PublicAPI.Unshipped.txt [hunk 2]
```diff
@@ -9,4 +9,4 @@ OpenTelemetry.Exporter.ConsoleExporterOutputTargets.Console = 1 -> OpenTelemetry
 OpenTelemetry.Exporter.ConsoleExporterOutputTargets.Debug = 2 -> OpenTelemetry.Exporter.ConsoleExporterOutputTargets
 OpenTelemetry.Trace.ConsoleExporterHelperExtensions
 override OpenTelemetry.Exporter.ConsoleExporter<T>.Export(in OpenTelemetry.Batch<T> batch) -> OpenTelemetry.ExportResult
-static OpenTelemetry.Trace.ConsoleExporterHelperExtensions.AddConsoleExporter(this OpenTelemetry.Trace.TracerProviderBuilder builder, System.Action<OpenTelemetry.Exporter.ConsoleExporterOptions> configure = null) -> OpenTelemetry.Trace.TracerProviderBuilder
\ No newline at end of file
+static OpenTelemetry.Trace.ConsoleExporterHelperExtensions.AddConsoleExporter(this OpenTelemetry.Trace.TracerProviderBuilder builder, System.Action<OpenTelemetry.Exporter.ConsoleExporterOptions> configure = null) -> OpenTelemetry.Trace.TracerProviderBuilder
```
Comment (cijothomas) [inline src/OpenTelemetry.Exporter.Console/.publicApi/net452/PublicAPI.Unshipped.txt:12]: nothing changed here?

**Comment by cijothomas** [comment]
@reyang @CodeBlanch please share your thoughts as well.

File: src/OpenTelemetry.Exporter.Console/.publicApi/net452/PublicAPI.Unshipped.txt (hunk 2 above)
Comment (alanwest) [reply src/OpenTelemetry.Exporter.Console/.publicApi/net452/PublicAPI.Unshipped.txt:12]: Newline was added at the end of the file. My editor probably did it automatically.

**Review by reyang — APPROVED** [review]
LGTM.

**Review by eddynaka — APPROVED** [review]

**Review by cijothomas — APPROVED** [review]

**Comment by CodeBlanch** [comment]
I definitely agree with the approach that (in most cases) extensions should live in the namespace of whatever is being extended. Thanks @alanwest!

**Pull request merged**